File: 250402/등차수열/arithmetic-sequence.cpp
#include <iostream>

#define MAX_N 100
#define MAX_K 100

using namespace std;

int N;
int arr[MAX_N];
int k_count[MAX_K + 1] = {0};
int k;

int main() {
    cin >> N;

    for (int i = 0; i < N; i++) {
        cin >> arr[i];
    }

    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            int total = arr[i] + arr[j];
            if (total % 2 != 0) {
                continue;
            }

            k = total / 2;
            if (k <= MAX_K) {
                k_count[k]++;
            }
        }
    }

    int max_count = 0;
    for (int i = 0; i <= MAX_K; i++) {
        if (k_count[i] > max_count) {
            max_count = k_count[i];
        }
    }

    cout << max_count;

    return 0;
}

File: 250402/등차수열/arithmetic-sequence.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] arr = new int[N];
        int k;
        int[] k_count = new int[101];

        for (int i = 0; i < N; i++) {
            arr[i] = sc.nextInt();
        }
        
        for (int i = 0; i < N; i++) {
            for (int j = i + 1; j < N; j++) {
                int total = arr[i] + arr[j];
                if (total % 2 != 0) {
                    continue;
                }

                k = total / 2;
                if (k <= 100) {
                    k_count[k]++;
                }
            }
        }

        int max_count = 0;
        for (int i = 0; i <= 100; i++) {
            if (k_count[i] > max_count) {
                max_count = k_count[i];
            }
        }

        System.out.printf("%d", max_count);
    }
}